numcodecs/msgpacks.py

```diff
@@ -0,0 +1,52 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import, print_function, division
+
+
+import numpy as np
+
+
+from numcodecs.abc import Codec
+from numcodecs.compat import ndarray_from_buffer, buffer_copy
+import msgpack
+
+
+class MsgPack(Codec):
+    """Codec to encode data as as msgpacked bytes. Useful for encoding python
+    strings
+
+    Raises
+    ------
+    encoding a non-object dtyped ndarray will raise ValueError
+
+    Examples
+    --------
+    >>> import numcodecs as codecs
+    >>> import numpy as np
+    >>> x = np.array(['foo', 'bar', 'baz'], dtype='object')
+    >>> f = codecs.MsgPack()
+    >>> f.decode(f.encode(x))
+    array(['foo', 'bar', 'baz'], dtype=object)
+
+    """  # flake8: noqa
+
+    codec_id = 'x-msgpack'
```

Comment:
This can be just 'msgpack'. I was using 'x-' as a prefix for codecs not provided by numcodecs itself, but now this codec is inside numcodecs we can drop this prefix..